src/libstd/sys/unix/fs.rs

```diff
@@ -132,6 +139,32 @@ impl FromInner<raw::mode_t> for FilePermissions {
 impl Iterator for ReadDir {
     type Item = io::Result<DirEntry>;
 
+    #[cfg(target_os = "sunos")]
```

Comment:
Does sunos not have `readdir_r`? In posix at least the `readdir` function isn't thread safe (I believe) and can be harmful to use, so it would be best to share implementations here if possible.